### ISSUE TITLE
Problem with async methods

### ISSUE THREAD

**Issue by kolesnikov-sergey** [lucaslorentz/minicover#51]
If the class contains async method, the reports are generated incorrectly (opencover, clover, etc)

# Example
### Async class
```
using System;
using System.Threading.Tasks;

namespace QnpMiddleService
{
    public class AsyncExample
    {
        public void DoNotAsync()
        {
            Console.WriteLine("Do Not Async");
        }

        public async Task DoAsync()
        {
            await Task.Delay(1);
            Console.WriteLine("Do Async");
        }

        public async Task DoAsyncAnother()
        {
             await Task.Delay(1);
            Console.WriteLine("Do Async another");
        }
    }
}
```

### opencover report
```
<?xml-stylesheet type='text/xsl' href='coverage.xsl'?>
<CoverageSession>
  <Module hash="353e72eb-ce92-4b3c-ac52-68e0b274e447">
    <FullName>QnpMiddleService</FullName>
    <ModuleName>QnpMiddleService</ModuleName>
    <File uid="1" fullPath="C:\Users\U_M0XZD\Projects\qnp\qnp-middle-service\src\QnpMiddleService\AsyncExample.cs" />
    <Class>
      <FullName>QnpMiddleService.AsyncExample</FullName>
      <Method visited="false" isConstructor="false">
        <Name>System.Void QnpMiddleService.AsyncExample::DoNotAsync()</Name>
        <FileRef uid="1" />
        <SequencePoint vc="0" uspid="1" ordinal="1" sl="10" sc="13" el="10" ec="47" />
      </Method>
    </Class>
    <Class>
      <FullName>QnpMiddleService.AsyncExample/&lt;DoAsync&gt;d__1</FullName>
      <Method visited="false" isConstructor="false">
        <Name>System.Void QnpMiddleService.AsyncExample/&lt;DoAsync&gt;d__1::MoveNext()</Name>
        <FileRef uid="1" />
        <SequencePoint vc="0" uspid="2" ordinal="1" sl="15" sc="13" el="15" ec="33" />
        <SequencePoint vc="0" uspid="3" ordinal="2" sl="15" sc="13" el="15" ec="33" />
        <SequencePoint vc="0" uspid="4" ordinal="3" sl="16" sc="13" el="16" ec="43" />
        <SequencePoint vc="0" uspid="5" ordinal="4" sl="16" sc="13" el="16" ec="43" />
      </Method>
    </Class>
    <Class>
      <FullName>QnpMiddleService.AsyncExample/&lt;DoAsyncAnother&gt;d__2</FullName>
      <Method visited="false" isConstructor="false">
        <Name>System.Void QnpMiddleService.AsyncExample/&lt;DoAsyncAnother&gt;d__2::MoveNext()</Name>
        <FileRef uid="1" />
        <SequencePoint vc="0" uspid="6" ordinal="1" sl="21" sc="14" el="21" ec="34" />
        <SequencePoint vc="0" uspid="7" ordinal="2" sl="21" sc="14" el="21" ec="34" />
        <SequencePoint vc="0" uspid="8" ordinal="3" sl="22" sc="13" el="22" ec="51" />
        <SequencePoint vc="0" uspid="9" ordinal="4" sl="22" sc="13" el="22" ec="51" />
      </Method>
    </Class>
  </Module>
</CoverageSession>
```

**Comment by Kantuz001** [comment]
@lucaslorentz can u point file and method so that we can fix it?

**Comment by lucaslorentz** [comment]
So, those 3 methods should be merged and considered the same method?
Maybe we can just convert the methodName and methodFullName to the original async method during instrumentation:
https://github.com/lucaslorentz/minicover/blob/master/src/MiniCover/Instrumentation/Instrumenter.cs#L268

Probabbly those generated method names have some pattern that we can rely on.

**Comment by lucaslorentz** [comment]
Maybe check how Ben Adams does it on: https://github.com/benaadams/Ben.Demystifier

**Comment by lucaslorentz** [comment]
@bhugot 
Do you agree about fixing the method names during instrumentation?
Another option would be do it during reporting, but I think during instrumentation is better.

**Comment by bhugot** [comment]
no we should do it during instrumentation and same thing with tests

**Comment by VijayakumarNatarajan** [comment]
Has this been fixed ?

**Comment by lucaslorentz** [comment]
Fixed on version v3.0.0-alpha-1
